tutorials/15-external-samplers/external-samplers.jmd

```diff
@@ -0,0 +1,104 @@
+---
+title: Using External Sampler
+permalink: /tutorials/:name/
+mathjax: true
+weave_options:
+  error : false
+---
+
+# Using External Samplers
+
+Turing currently wraps some of the most popular sampling libraries in the Julia ecosystem.
+These wrappers allow new users to seemlessly sample statistical models without leaving Turing.
+However, these wrappers often lack the latests functionalities implemented on the sampling libraries that more experienced users might want to use.
+Moreover, users might want to use samplers currently not wrapped within Turing.
+
+For these reasons, Turing now allows users the possibility of defining their own sampler and using it to sample a Turing model.
+Let us begin by defining a simple Turing model to showcase the use of external samplers.
+For this demonstration we will use a 10 dimensional Neal's funnel:
+
+```julia
+# Import libraries.
+using Turing, Random, LinearAlgebra
+
+d = 10
+@model function funnel()
+    θ ~ Truncated(Normal(0, 3), -3, 3)
+    z ~ MvNormal(zeros(d - 1), exp(θ) * I)
+    return x ~ MvNormal(z, I)
+end
+```
+
+Let us know sample the model to generate some observations and condition our model on them:
+
+```julia
+Random.seed!(1)
+(; x) = rand(funnel() | (θ=0,))
+model = funnel() | (; x)
+```
+
+Users can now use any sampler algorithm they wish to sample this model as long as it is of the type `AbstractMCMC.AbstractSampler`.
+Before discussing how is this in done in practice, it is interesting to give a high level description of the process.
+Imagine that we created an instance of external sampler that we will call `spl` such that `typeof(spl)<:AbstractMCMC.AbstractSampler`.
+In order to avoid type ambiguity within Turing, at the moment is necessary to declare `spl` as an external sampler to Turing this is done as:
```

Comment:
I don't think this is particularly interesting to the user :confused: 

Though I do agree it might be of interest to a small minority, so might be nice to keep in. _But_ then I'd suggest moving to later in the tutorial as a description of why we have to use`externalsampler`, etc.

Reply:
How would you convey to the user the current need to wrap external samplers?
I am not married to the current text but I think that should be mentioned.